source/neuropod/internal/neuropod_tensor.cc

```diff
@@ -17,12 +17,25 @@ limitations under the License.
 
 #include "neuropod/internal/error_utils.hh"
 
+#include <algorithm>
+
 namespace neuropod
 {
 
 namespace
 {
 
+size_t compute_num_elements(const std::vector<int64_t> &dims)
+{
+    // Get the number of elements in the tensor by multiplying all the dims together
+    return std::accumulate(dims.begin(), dims.end(), static_cast<size_t>(1), std::multiplies<>());
```

Comment:
Not critical, I left comments in prev iteration, exact headers for these algos are:

#include <functional> // std:: multiplies
#include <numeric> // std::accumulate

Reply:
Ah I forgot to submit my response to your comments. Based on [`include-what-you-use`](https://include-what-you-use.org/) from Google:
> "Include what you use" means this: for every symbol (type, function variable, or macro) that you use in foo.cc, either foo.cc or foo.h should #include a .h file that exports the declaration of that symbol. 

For `std::accumulate`, my updates did add `#include <algorithm>` to the `.cc` file

For `std::multiplies`, `neuropod_tensor.hh` includes `<functional>`

Reply:
_Note: after looking just now, the include-what-you-use definition contradicts the [Google style guide recommendation](https://google.github.io/styleguide/cppguide.html#Include_What_You_Use):_

> ... foo.cc should include bar.h if it uses a symbol from it even if foo.h includes bar.h. ...

That said, I think it's okay for us to stick to the definition in my comment above. If we introduce IWYU into lint/CI, we can enforce it throughout the codebase.

Reply:
Tracked by #446

Reply:
Just found that github formatter hided header name because of not escaped \<. Just minor note:

https://en.cppreference.com/w/cpp/algorithm/accumulate "Defined in header <numeric>" , I guess that "algorithm" includes it as well and this is why it works and is under the same "include-what-you-use" topic you mentioned.

Thanks for creating Issue.